### PR TITLE
[VL] Fix the issue of SubstraitToVeloxPlanConverter::createNotEqualFilter

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

we will create two range filters for `createNotEqualFilter`, the `lowerFilter` and the `upperFilter`, the`NotEqual` will be transferred to `[minLower, value) || (value, maxUpper]`.

the `lowerFilter` represents the range `(value, maxUpper]`, and the `upperFilter` represents the range `[minLower, value)`.

then add `lowerFilter` first and add `upperFilter` second, which makes the multi-range filters violate the check rules in `BigintMultiRange::BigintMultiRange()`.

we can swap the adding to fix it.

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Review by rui-mo — COMMENTED** [review]
Thanks! Just several tiny issues.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 1]
```diff
@@ -1118,7 +1113,7 @@ connector::hive::SubfieldFilters SubstraitToVeloxPlanConverter::createSubfieldFi
     columnToFilterInfo[idx];
   }
 
-  // Construct the FilterInfo for the related column.
+  // process scalarFunctions
```
[the remark this answers is not in the document]
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1116]: OK

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 2]
```diff
@@ -1605,7 +1596,10 @@ void SubstraitToVeloxPlanConverter::createNotEqualFilter(
       false, /*upperUnbounded*/
       true, /*upperExclusive*/
       nullAllowed); /*nullAllowed*/
+
+  // keep this emplace_back order
```
[the remark this answers is not in the document]
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1600]: OK

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 3]
```diff
@@ -821,11 +819,8 @@ core::PlanNodePtr SubstraitToVeloxPlanConverter::toVeloxPlan(const ::substrait::
 
     // Separate the filters to be two parts. The subfield part can be
     // pushed down.
-    std::vector<::substrait::Expression_ScalarFunction> subfieldFunctions;
-    std::vector<::substrait::Expression_SingularOrList> subfieldrOrLists;
-
-    std::vector<::substrait::Expression_ScalarFunction> remainingFunctions;
-    std::vector<::substrait::Expression_SingularOrList> remainingrOrLists;
+    std::vector<::substrait::Expression_ScalarFunction> subfieldFunctions, remainingFunctions;
+    std::vector<::substrait::Expression_SingularOrList> subfieldrOrLists, remainingrOrLists;
```
[the remark this answers is not in the document]
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:823]: yes, I will revert changes here

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 4]
```diff
@@ -1824,11 +1821,10 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
       upperBound = getMax<NativeType>();
     }
 
-    bool lowerUnbounded = true;
-    bool upperUnbounded = true;
-    bool lowerExclusive = false;
-    bool upperExclusive = false;
+    bool lowerUnbounded = true, upperUnbounded = true;
+    bool lowerExclusive = false, upperExclusive = false;
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1825]: Do we need to revert this change also?

Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc]: done

**Review by rui-mo — APPROVED** [review]
